src/FullBuild/Baselines.fs

```diff
@@ -61,84 +62,131 @@ with
+                                    |> not) 
+            |> Set.ofSeq
+        Set.union dif overrideWith
+
+    member this.GetPulledBaseline () : Baseline option =
+        let baselineFileInfo = Env.GetBaselineFile() 
+        if baselineFileInfo.Exists then
+            baselineFileInfo.FullName |> parseBaselineFile |> Some
+        else
+            None
+
+    member this.GetSourcesBaseline () : Baseline =
+        let pulledBaseline = this.GetPulledBaseline()
+        let sourcesBookmarks = getClonedReposBookmarks ()
+
+        let resultBaseline = 
+            match pulledBaseline with
+            | Some baseline -> updateBookmarks baseline.Bookmarks sourcesBookmarks
+            | None -> sourcesBookmarks
+
         let branch = Configuration.LoadBranch()
-        let tagFilter = sprintf "fullbuild/%s/*" branch
-        match Tools.Vcs.FindLatestMatchingTag wsDir graph.MasterRepository tagFilter with
-        | Some tag -> let tagInfo = TagInfo.Parse tag
-                      Baseline(graph, tagInfo, false)
-        | _ -> let tagInfo = { TagInfo.BuildBranch = branch; TagInfo.BuildNumber = "temp" }
-               Baseline(graph, tagInfo, true)
-
-    member this.CreateBaseline (buildNumber : string) : Baseline =
-        let globals = Configuration.LoadGlobals()
-        let antho = Configuration.LoadAnthology()
-        let graph = Graph.from globals antho
+        let buildInfo = { BuildInfo.BuildBranch = branch; BuildInfo.BuildNumber = "1.0.0" }
+        Baseline(buildInfo, resultBaseline) 
+
+    member this.GetBinariesBaseline () : Baseline option =
+        let baselineFileInfo = Env.GetTemporaryBaselineFile()
+        if baselineFileInfo.Exists then
+            baselineFileInfo.FullName |> parseBaselineFile |> Some
+        else
+            None
+               
+    member this.UpdateBaseline (buildNumber : string) : unit =
         let branch = Configuration.LoadBranch()
-        let tagInfo = { TagInfo.BuildBranch = branch; TagInfo.BuildNumber = buildNumber }
-        Baseline(graph, tagInfo, true)
+        let buildInfo = { BuildInfo.BuildBranch = branch; BuildInfo.BuildNumber = buildNumber }
+
+        let wsDir = Env.GetFolder Env.Folder.Workspace
+
+        let sourcesBasline = this.GetSourcesBaseline()
+        let baseline = Baseline(buildInfo, sourcesBasline.Bookmarks)
+        let baselineFile = serializeBaseline baseline
+        Env.GetTemporaryBaselineFile().FullName |> baselineFile.Save
 
+     member this.FindMatchingBuildInfo () : BuildInfo option =
+        let branch = Configuration.LoadBranch()
+        let tagFilter = sprintf "fullbuild/%s/*" branch
+        match Tools.Vcs.FindLatestMatchingTag wsDir graph.MasterRepository tagFilter with
+        | Some tag -> BuildInfo.Parse tag |> Some
+        | _ -> None
+        
+    member this.TagMasterRepository (comment : string) : unit =
+        let wsDir = Env.GetFolder Env.Folder.Workspace
+        let baseline = this.GetPulledBaseline() |> Option.get
```

Comment:
In this case before "history" and "view --modified" commands the user should use "pull --bin" command (if the build was done before)